packages/orchestration/test/cosmos-interchain-service.test.ts

```diff
@@ -0,0 +1,397 @@
+import { test } from '@agoric/zoe/tools/prepare-test-env-ava.js';
+import { toRequestQueryJson } from '@agoric/cosmic-proto';
+import {
+  QueryBalanceRequest,
+  QueryBalanceResponse,
+} from '@agoric/cosmic-proto/cosmos/bank/v1beta1/query.js';
+import {
+  MsgDelegate,
+  MsgDelegateResponse,
+} from '@agoric/cosmic-proto/cosmos/staking/v1beta1/tx.js';
+import { Any } from '@agoric/cosmic-proto/google/protobuf/any.js';
+import { matches } from '@endo/patterns';
+import { heapVowE as E } from '@agoric/vow/vat.js';
+import { decodeBase64 } from '@endo/base64';
+import type { LocalIbcAddress } from '@agoric/vats/tools/ibc-utils.js';
+import { getMethodNames } from '@agoric/internal';
+import type { Port } from '@agoric/network';
+import { eventLoopIteration } from '@agoric/internal/src/testing-utils.js';
+import type { IBCMethod } from '@agoric/vats';
+import { commonSetup } from './supports.js';
+import { ChainAddressShape } from '../src/typeGuards.js';
+import { tryDecodeResponse } from '../src/utils/cosmos.js';
+import { buildChannelCloseConfirmEvent } from '../tools/ibc-mocks.js';
+
+const CHAIN_ID = 'cosmoshub-99';
+const HOST_CONNECTION_ID = 'connection-0';
+const CONTROLLER_CONNECTION_ID = 'connection-1';
+
+test.serial('makeICQConnection returns an ICQConnection', async t => {
+  const {
+    bootstrap: { cosmosInterchainService },
+  } = await commonSetup(t);
+
+  const icqConnection = await E(cosmosInterchainService).provideICQConnection(
+    CONTROLLER_CONNECTION_ID,
+  );
+  const [localAddr, remoteAddr] = await Promise.all([
+    E(icqConnection).getLocalAddress(),
+    E(icqConnection).getRemoteAddress(),
+  ]);
+  t.log(icqConnection, {
+    localAddr,
+    remoteAddr,
+  });
+  t.regex(localAddr, /ibc-port\/icqcontroller-\d+/);
+  t.regex(
+    remoteAddr,
+    new RegExp(`/ibc-hop/${CONTROLLER_CONNECTION_ID}`),
+    'remote address contains provided connectionId',
+  );
+  t.regex(
+    remoteAddr,
+    /icqhost\/unordered\/icq-1/,
+    'remote address contains icqhost port, unordered ordering, and icq-1 version string',
```

Comment:
nice tests

Reply:
Thanks, although this was an existing test (appearing because of the file rename). 

See L232-EOF for the 2 new tests.